extensions/objects/templates/ImageWithRegionsEditor.js

```diff
@@ -192,6 +206,48 @@ oppia.directive('imageWithRegionsEditor', [
             height: (area[1][1] - area[0][1]) * $scope.getImageHeight()
           };
         };
+        var resizeRegion = function() {
+          var labeledRegions = $scope.$parent.value.labeledRegions;
+          var resizedRegion = labeledRegions[$scope.selectedRegion].region;
+          var deltaX = $scope.mouseX - $scope.originalMouseX;
+          var deltaY = $scope.mouseY - $scope.originalMouseY;
+          var x = $scope.originalRectArea.x;
+          var y = $scope.originalRectArea.y;
+          var width = $scope.originalRectArea.width;
+          var height = $scope.originalRectArea.height;
+          if (height - $scope.yDirection * deltaY <= 0 &&
+              $scope.yDirectionToggled === 0) {
+            $scope.yDirectionToggled = 1;
+          } else if (height - $scope.yDirection * deltaY >= 0 &&
+            $scope.yDirectionToggled === 1) {
+            $scope.yDirectionToggled = 0;
+          }
+          if ($scope.yDirection === 1) {
+            y = y + !$scope.yDirectionToggled * deltaY +
```

Comment:
It is odd to use ! in this context, since that results in a boolean. If you want the $scope.yDirectionToggled vars to be booleans, define them as such and do something like:

    y += $scope.yDirectionToggled ? height : deltaY;

Ditto below.